### PR TITLE
ROC-6769 Notification for successful mediation

### PR DESCRIPTION
**Pull request by Valkrys**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-6769

### Change description ###
Create notification for both parties in case of ccd event successful mediation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandler.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/MediationSuccessfulNotificationService.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516488&bid=15216163)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/MediationSuccessfulNotificationService.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516488&bid=15216163)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulNotificationServiceTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandler.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulNotificationService.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandlerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516488&bid=15216163)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516488&bid=15216163)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516488&bid=15216163)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandler.java  1
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandlerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516488&bid=15216163)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516488&bid=15216163)

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ClaimServiceTest.java [hunk 1]
```diff
@@ -279,20 +279,6 @@ public void requestMoreTimeToRespondShouldThrowNotFoundExceptionWhenClaimNotFoun
...
-        LocalDate responseDeadlineInThePast = now()
-            .minusDays(10);
-        Claim claim = createClaimModel(responseDeadlineInThePast, false);
-
-        when(caseRepository.getClaimByExternalId(eq(EXTERNAL_ID), any()))
-            .thenReturn(Optional.of(claim));
-        when(userService.getUser(eq(AUTHORISATION))).thenReturn(USER);
-
-        claimService.requestMoreTimeForResponse(EXTERNAL_ID, AUTHORISATION);
-    }
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/ClaimServiceTest.java]: Why is this removed? Seems unrelated

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+        verify(notificationService).sendMail(eq(claim.getDefendantEmail()),
+                eq(MEDIATION_SUCCESSFUL_DEFENDANT),
+                any(),
+                eq("to-defendant-mediation-successful"));
+
+        verify(notificationService).sendMail(eq(claim.getSubmitterEmail()),
+                eq(MEDIATION_SUCCESSFUL_CLAIMANT),
+                any(),
+                eq("to-claimant-mediation-successful"));
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandlerTest.java]: There are two separate tests here; recommend splitting them into two `@Test` methods

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/NotificationReferenceBuilder.java [hunk 3]
```diff
@@ -218,4 +218,22 @@ public static String referenceForClaimant(String claimReferenceNumber) {
             return referenceForClaim(claimReferenceNumber, CLAIMANT);
         }
     }
+
+    public static class MediationSuccessful {
+
+        public static final String TEMPLATE = "to-%s-mediation-successful";
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/NotificationReferenceBuilder.java]: Does this need to be exposed? Change to `private` if not

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ClaimServiceTest.java (hunk 1 above)
Comment (Valkrys) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/ClaimServiceTest.java]: I don't know how to undo changes to this file so I deleted it because it is deleted in another pr too.

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandlerTest.java (hunk 2 above)
Comment (Valkrys) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandlerTest.java]: okay

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/NotificationReferenceBuilder.java (hunk 3 above)
Comment (Valkrys) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/notifications/NotificationReferenceBuilder.java]: I am not sure, Ellie said this has something to do with gov notify

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandler.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    @Override
+    protected Map<CallbackType, Callback> callbacks() {
+        return ImmutableMap.of(
+                CallbackType.SUBMITTED, this::notifyParties
+        );
+    }
+
+    @Override
+    public List<CaseEvent> handledEvents() {
+        return ImmutableList.of(MEDIATION_SUCCESSFUL);
```
[the remark this answers is not in the document]
Comment (Valkrys) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandler.java]: oke

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516488&bid=15216163)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandler.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516488&bid=15216163)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandler.java  2
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandlerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516488&bid=15216163)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandler.java  2
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandlerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516488&bid=15216163)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/mediation/MediationSuccessfulCallbackHandler.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/cmc-claim-store/cmc-claim-store/pullRequest?prid=4516488&bid=15216163)
